code/modules/emotes/custom_emote.dm

```
@@ -0,0 +1,87 @@
+
+/// This is the custom_emote that you'll want to use if you want the mob to be able to input their emote.
+/mob/proc/custom_emote(var/m_type = VISIBLE_MESSAGE, var/message, var/range = world.view, var/check_stat = TRUE)
+
+	if((check_stat && (usr && stat)) || (!use_me && usr == src))
+		to_chat(src, "You are unable to emote.")
+		return
+
+	var/input
+	if(!message)
+		input = sanitize(tgui_input_text(src,"Choose an emote to display."))
+	else
+		input = message
+	process_emote(m_type, message, input, range)
+
+/// This is the custom_emote that you'll want to use if you're forcing something to custom emote with no input from the mob.
+/// By default, we have a visible message, our range is world.view, and we do NOT check the stat.
+/mob/proc/automatic_custom_emote(var/m_type = VISIBLE_MESSAGE, var/message, var/range = world.view, var/check_stat = FALSE)
+	if(check_stat && (usr && stat))
+		to_chat(src, "You are unable to emote.")
```

do we want to message users on failures? Or just let automatic emotes fail silently?

---

I made it fail silently. No need to be told you can't emote when your liver is acting up and you can't scream because you're in an alcohol coma